src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -145,81 +147,32 @@ public unsafe BitArray(bool[] values)
             // (true for any non-zero values, false for 0) - any values between 2-255 will be interpreted as false.
             // Instead, We compare with zeroes (== false) then negate the result to ensure compatibility.
 
-            if (Avx2.IsSupported)
+            ref byte value = ref Unsafe.As<bool, byte>(ref MemoryMarshal.GetArrayDataReference<bool>(values));
```

Comment:
Do `VectorXx<T>` not support T == bool?

Reply:
No and supporting `bool` directly might actually cause downstream confusion due to how users expect operations to work in C# vs how operations actually work in IL.

This makes it explicit we are interpreting data as `byte` and need to account for ILs `0` vs `non-zero` (rather than C#'s `0` vs `1` vs `undefined`)